collector/src/main.rs

```diff
@@ -585,6 +664,37 @@ fn main_result() -> anyhow::Result<i32> {
             (@arg RUSTDOC: --rustdoc +takes_value "The path to the local rustdoc to benchmark")
         )
 
+        (@subcommand diff_local =>
+            (about: "Profiles and compares two toolchains with one of several profilers")
+
+            // Mandatory arguments
+            (@arg PROFILER: +required +takes_value
+             "One of: 'self-profile', 'time-passes', 'perf-record',\n\
+             'oprofile', 'cachegrind', 'callgrind', 'dhat', 'massif',\n\
+             'eprintln', 'llvm-lines'")
+            (@arg RUSTC1:   +required +takes_value "The path to the local rustc to benchmark")
+            (@arg ID1:      +required +takes_value "Identifier to associate benchmark results with")
+            (@arg RUSTC2:   +required +takes_value "The path to the local rustc to benchmark")
+            (@arg ID2:      +required +takes_value "Identifier to associate benchmark results with")
```

Comment:
Can we skip the IDs? Or maybe make them optional (that may be hard).

I generally diff locally across downloaded commits and it (was/would be) annoying to type e.g. before after. We can use the rustc1 and rustc2 parameters as IDs if they're in the +foo toolchain syntax and fallback on before/after otherwise, I think.